main.c

```diff
@@ -1,3 +1,44 @@
+#include <stdint.h>
+
+#include "am1808.h"
+
+// Shared RAM (control input)
+typedef struct shared_ram {
+    // Declare this as a u32 to force more-efficient codegen
+    uint32_t pwms;
+} shared_ram;
+static volatile shared_ram * const SHARED = (volatile shared_ram *)0x80010000;
+// XXX this doesn't seem to work (compiler won't generate code to use it)
+// #pragma ctable_entry 30 0x80010000
+
+// LED definitions (all on GP6)
+#define LED0    (1 << 13)
+#define LED1    (1 << 7)
+#define LED2    (1 << 14)
+#define LED3    (1 << 12)
+
+static inline void update_pwm(uint8_t val, uint8_t time_now, uint32_t gpio_bit) {
+    if (time_now < val) {
+        GPIO->set_data67 = gpio_bit;
+    } else {
+        GPIO->clr_data67 = gpio_bit;
+    }
+}
+
 void main() {
-    __halt();
+    uint32_t pwms = 0;
+    while (1) {
+        // 24 MHz / 256 ==> 93.75 kHz tick rate for this counter
+        uint8_t time_now = (*TIMER0_TIM34) >> 8;
+
+        if (time_now == 0) {
+            // 24 MHz / 256 / 256 ~= 366 Hz update rate
+            pwms = SHARED->pwms;
```

Comment:
```suggestion
            #pragma ctable_entry 30 0x80010000
            pwms = *(uint32_t *)0x80010000;
```

Does this work? (Obviously, we'll need to set the `C30` index in `CONTABPROPTR1` for this.

Reply:
It doesn't, and I believe I've found a compiler bug. See the updated comment.